alacritty/src/window.rs

```diff
@@ -386,6 +412,95 @@ impl Window {
     }
 }
 
+#[cfg(not(any(target_os = "macos", target_os = "windows")))]
+#[derive(Debug, Clone)]
+struct AlacrittyWaylandTheme {
+    background: Rgb,
+    foreground: Rgb,
+    dim_foreground: Rgb,
+    hovered_close_icon: Rgb,
+    hovered_maximize_icon: Rgb,
+    hovered_minimize_icon: Rgb,
+}
+
+#[cfg(not(any(target_os = "macos", target_os = "windows")))]
+impl WaylandTheme for AlacrittyWaylandTheme {
+    fn primary_color(&self, _window_active: bool) -> [u8; 4] {
+        [0xff, self.background.r, self.background.g, self.background.b]
+    }
+
+    fn secondary_color(&self, window_active: bool) -> [u8; 4] {
+        if window_active {
+            [0xff, self.foreground.r, self.foreground.g, self.foreground.b]
+        } else {
+            [0xff, self.dim_foreground.r, self.dim_foreground.g, self.dim_foreground.b]
+        }
+    }
+
+    fn close_button_color(&self, _status: WaylandThemeButtonState) -> [u8; 4] {
+        [0x00, self.background.r, self.background.g, self.background.b]
+    }
+
+    fn close_button_icon_color(&self, status: WaylandThemeButtonState) -> [u8; 4] {
+        match status {
+            WaylandThemeButtonState::Hovered => [
+                0xff,
+                self.hovered_close_icon.r,
+                self.hovered_close_icon.g,
+                self.hovered_close_icon.b,
+            ],
+            WaylandThemeButtonState::Idle => {
+                [0xff, self.foreground.r, self.foreground.g, self.foreground.b]
+            },
+            WaylandThemeButtonState::Disabled => {
+                [0xff, self.dim_foreground.r, self.dim_foreground.g, self.dim_foreground.b]
+            },
+        }
```

Comment:
I wonder if we could maybe write a little helper for that which would shorten things down?

Like `self.button_icon_color(status, self.hovered_close_icon)`.

Could then use that for all three buttons with just a single line in each of the trait implementations.